weaviate/replication/models.py

```diff
@@ -0,0 +1,144 @@
+import uuid
+from dataclasses import dataclass
+from enum import Enum
+from typing import Generic, List, Literal, Optional, TypedDict, TypeVar
+
+
+class TransferType(str, Enum):
+    """Enum for transfer types."""
+
+    COPY = "COPY"
+    MOVE = "MOVE"
+
+
+class ReplicateOperationState(str, Enum):
+    """Enum for replication operation states."""
+
+    REGISTERED = "REGISTERED"
+    HYDRATING = "HYDRATING"
+    FINALIZING = "FINALIZING"
+    DEHYDRATING = "DEHYDRATING"
+    READY = "READY"
+    CANCELLED = "CANCELLED"
+
+
+class _ReplicationReplicateDetailsReplicaStatus(TypedDict):
+    state: Literal["REGISTERED", "HYDRATING", "FINALIZING", "DEHYDRATING", "READY", "CANCELLED"]
```

Comment:
can't you just use the enum as type here?

Reply:
These are just the types of the API response so are the types of `response.json()`. The `Literal` here is for internal use only to avoid serialization. Turning them into enums would just move any error on incompatible types from the public class to this private one

Reply:
It is more about avoiding duplication - I am sure there will be more states added and then you have to add it in two places (which will be forgotten)

Reply:
Ideally, these `TypedDict` classes would be auto-generated from the `schema.json` file, but I don't think such a thing exists. The duplication is a good point, I will simply remove these `TypedDict` classes and have the serialisation work on raw `dict`. Either way, it needs runtime tests to ensure correctness anyway despite `pyright`